lib/src/query.c

```diff
@@ -3541,8 +3542,8 @@ static inline bool ts_query_cursor__advance(
         point_gte(ts_node_start_point(parent_node), self->end_point)
       );
       bool node_precedes_range = parent_precedes_range || (
-        ts_node_end_byte(node) <= self->start_byte ||
-        point_lte(ts_node_end_point(node), self->start_point)
+        (!is_empty && ts_node_end_byte(node) <= self->start_byte) ||
+        (!is_empty && point_lte(ts_node_end_point(node), self->start_point))
```

Comment:
Maybe it's clearer if you factor `is_empty` out of those conditions
```suggestion
        !is_empty && (
          ts_node_end_byte(node) <= self->start_byte ||
          point_lte(ts_node_end_point(node), self->start_point))
```

Reply:
totally agree, thanks